### DISCUSSION TITLE
Feature proposal: use more than one folder in the index/project

### DISCUSSION THREAD

**Discussion by quesada** [zim-desktop-wiki/zim-desktop-wiki#1806] [Ideas]
Right now a notebook is a folder.
This has one downside: if I share it with someone, it's all or nothing. In my company we have teams; each team could have a folder, and my zim (I'm supervising everything) should let me index all these folders under one single notebook (or zim window). I can have multiple notebooks open right now, but each is its own window. Not ideal. Not easy to search across them. Does this make sense?

**Comment by jaap-karssenberg** [comment]
I would like the application to work well with multiple notebooks. Adding
everything (be it two team, be it hobby vs work, ...) in one big notebook
does not conceptually match with how you would use paper notebooks. So my
first question would be what changes would enable you to work well with two
notebooks. E.g. a multi-notebook search is not that hard to create and does
not require the notebooks to share a single index.

However if you must, you could try setting up a notebook with sub-sections
per team and share those sub-folders per team. When these are opened as
notebooks on their own everything *should* still work. Under Linux you can
use symlinks to link folders if the shares are in different locations.

Regards,

Jaap

**Comment by quesada** [comment]
Hi Jaap,
I tried running with this solution (two subfolders, shared with different people). It's problematic as soon as you get 'out' of zim. For example, clicking on a txt file on the file manager would recreate the index using the 'parent folder' as root, then breaking the idea of having one index for both.

I'm trying to use recoll to index zim notes, but clicking on a result there produces this as well.

**Reply by introt** [reply]
Hi! Moved this to discussions as per the issue list cleanup plan ("proposals should go to discussions first and turn into an issue ticket when someone is actually working on it").

> For example, clicking on a txt file on the file manager would recreate the index using the 'parent folder' as root

To elaborate: Zim chooses the closest (fewest levels up) `notebook.zim` it can find; this can be countered by simply making your subnotebook's `notebook.zim` files unavailable. Options include:
* having your teams exclude `notebook.zim` from the synced files
* deleting the extra `notebook.zim` files from your computer (and not syncing that change to theirs)
* patching your Zim to look for `mainbook.zim` instead
* filesystem trickery
* etc...

**Reply by quesada** [reply]
Yeah, it's a complex topic. We are basically trying ot recreate 'permission based' access to notes, which other notetaking tools using databases instead of files can do easier. Not trivial in any case. Moving from 'single player mode' to team zim is quite complex, but I'd argue really needed in 2022.

**Reply by introt** [reply]
I'd counter-argue "team zim" is out-of-scope, but also congratulate you and your team for exercising [freedom 0](https://www.gnu.org/philosophy/free-sw.html#run-the-program). Please do document your unusual workflow, it sounds really interesting!
